### ISSUE TITLE
CA5401 false positive when SymmetricAlgorithm.CreateEncryptor is used in combination with key deriviation

### ISSUE THREAD

**Issue by avivanoff** [dotnet/roslyn-analyzers#3002]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.7 (Latest)

#### Diagnostic ID

CA5401

#### Repro steps

```
using (var keyGen =  = new Rfc2898DeriveBytes(pwd, salt, 100000))
using (var aes = Aes.Create())
using (var encryptor = aes.CreateEncryptor(keyGen.GetBytes(32), keyGen.GetBytes(16)))
{
    ...
}
```

#### Expected behavior
CA5401 should not be reported.

#### Actual behavior
CA5401 is reported.

**Comment by daviddunson** [comment]
There is also, no way to persist the key and IV and later use the persisted key and IV for decryption without suppressing this warning:

```
using var aes = new AesCryptoServiceProvider();
using var transform = aes.CreateEncryptor();
var encrypted = transform.TransformFinalBlock(value, 0, value.Length);
var key = aes.Key;
var iv = aes.IV;
```

and then later:

```
using var aes = new AesCryptoServiceProvider { Key = key, IV = iv };
using var transform = aes.CreateDecryptor(); // generates CA5401
var decrypted = transform.TransformFinalBlock(value, 0, value.Length);
```

or:

```
using var aes = new AesCryptoServiceProvider();
using var transform = aes.CreateDecryptor(key, iv); // generates CA5401
var decrypted = transform.TransformFinalBlock(value, 0, value.Length);
```
